BlockEQ/View Controllers/Settings/SettingsViewController.swift

```diff
@@ -94,21 +94,14 @@ extension SettingsViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let settingSection = optionList[indexPath.section]
         let settingNode = settingSection.subnode(row: indexPath.row)
+        let cell = settingSection.cell(for: indexPath, tableView: tableView, viewController: self)
 
-        let cell = settingSection.cell(for: indexPath.row, tableView: tableView, viewController: self)
-        cell.selectionStyle = .none
-        cell.textLabel?.textColor = .gray
-
-        // If a setting node returns nil for a subnode - it's a leaf option and doesn't need a disclosure indicator
-        if let setting = settingNode, setting.leaf {
-            cell.accessoryType = .none
-        } else {
-            cell.accessoryType = .disclosureIndicator
-        }
-
-        if settingNode?.enabled ?? false {
-            cell.selectionStyle = .default
-            cell.textLabel?.textColor = .black
+        if let setting = settingNode, let settingCell = cell as? UpdatableCell {
+            settingCell.update(for: setting)
```

Comment:
@satbam This allows us to subclass `UITableViewCell` and have that subclass conform to `UpdatableCell`, which has a function that can be used to define how to set properties on that cell from within the subclass. 

This cleans up the usual bloat inside these tableview delegate methods.

Reply:
Fabulous - Nicely done!